src/js/drop.js

```diff
@@ -0,0 +1,128 @@
+const SECTORS = [SECTOR_1, SECTOR_2, SECTOR_3, SECTOR_4, SECTOR_5];
+const RED_COLOR = '#FF0000';
+const OPERATIONS = ['+', '-', '*', '/' ];
+export class Drop {
+    constructor(game, dropId, isSuperDrop = false, difficulty) {
+        this.game = game;
+        this.dropId = dropId;
+        this.isSuperDrop = isSuperDrop;
+        this.dropClass = this.isSuperDrop ? dropClasses.SUN : dropClasses.DROP;
+        this.difficulty = difficulty;        
+        this.minNumber = 0;
+        this.maxNumber = 10 + 10 * Math.trunc(dropId / 3);
+        this.number1 = this.getRandomNumber();
+        this.number2 = this.getRandomNumber();
+        this.operation = OPERATIONS[Math.round(Math.random() * this.difficulty * (OPERATIONS.length - 1))];        
+        this.sector = SECTORS[Math.round(Math.random() * (SECTORS.length - 1))];
+        this.dropElement = null;
+        this.isDestroyed = false;
+    }
+
+    static animate({timing, draw, duration}) {
+        let start = performance.now();
+      
+        return new Promise((resolve) => {
+            requestAnimationFrame(function animate(time) {
+                let timeFraction = (time - start) / duration;
+                if (timeFraction > 1) timeFraction = 1;  
+                
+                let progress = timing(timeFraction);
+            
+                draw(progress);
+            
+                if (timeFraction < 1) {
+                    requestAnimationFrame(animate);
+                } else {
+                    resolve();
+                }   
+            });
+        });
+    }
+
+    static isIntegerNumber(num) {
+        return (num ^ 0) === num;
+    }
+
+    getRandomNumber() {
+        return Math.round(Math.random() * (this.maxNumber - this.minNumber) + this.minNumber);
+    }
+
+    get result() {
+        const operationResult = eval(`${this.number1} ${this.operation} ${this.number2}`);
+
+        return Drop.isIntegerNumber(operationResult) 
+        ? operationResult 
+        : Math.round(operationResult);
+    }
+
+    createDropElement() {
+        const dropElement = `<div id="drop-${this.dropId}" class="${this.dropClass}">
+            <div class="operation-container">
+                <span class="operation">${this.operation}</span>
+            </div>
+            <div class="numbers-container">
+                <span class="number-1">${this.number1}</span>
+                <span class="number-2">${this.number2}</span>
+            </div>
+        </div>`;
+
+        this.sector.insertAdjacentHTML('afterbegin', dropElement);
+        this.dropElement = document.querySelector(`#drop-${this.dropId}`);
+    }
+
+    kill() {
+        this.dropElement.remove();
+        this.isDestroyed = true; 
+    }
+
+    destroy(success = false) {
+        if (this.isDestroyed) {
+            return;
+        }
+        this.game.notifyDestroyDrop(this, success);
+        
+        this.isDestroyed = true;
```

Comment:
please recheck is it really needed
I can see the same code line in kill method

Reply:
fixed